### PR TITLE
Avoid use of `from __future__ print_function`

### PR DESCRIPTION
**Pull request by niboshi**
`from __future__ print_function` can always be replaced with `fileobj.write`.
For Chainer code, it's better to be consistent.

For examples, `from __future__ print_function` may be confusing for beginners.

### CODE REVIEW

**Comment by kmaehashi** [comment]
Could you please confirm code style failures?

**Comment by niboshi** [comment]
Fixed.

**Comment by kmaehashi** [comment]
There are still some errors.
```
./examples/modelzoo/evaluate_caffe_net.py:58:59: E901 SyntaxError: invalid syntax
./examples/ptb/train_ptb_custom_loop.py:126:80: E501 line too long (80 > 79 characters)
```

**Comment by niboshi** [comment]
Thanks.

>./examples/modelzoo/evaluate_caffe_net.py:58:59: E901 SyntaxError: invalid syntax

For this error, I think it's better to remove `file=sys.stderr` than using `sys.stderr.write` for consistency with other examples.
How do you think?

**Comment by kmaehashi** [comment]
I agree. It seems there are no strong meaning to use stderr.
https://github.com/chainer/chainer/pull/110/commits/a73b91cb07fda21ac8c4e02cce5d3e30b487b5bd

**Comment by niboshi** [comment]
Thanks, fixed as such.

File: chainer/function_hooks/debug_print.py [hunk 1]
```diff
@@ -50,13 +49,14 @@ class PrintHook(function_hook.FunctionHook):
     name = 'PrintHook'
 
     def __init__(self, sep='', end='\n', file=sys.stdout, flush=True):
+        # TODO(niboshi): deprecate sep
```
Comment (kmaehashi) [inline chainer/function_hooks/debug_print.py]: Let's remove this line as https://github.com/chainer/chainer/pull/4471 got merged.

File: examples/modelzoo/evaluate_caffe_net.py [hunk 2]
```diff
@@ -124,8 +122,7 @@ def forward(x, t):
         del x, y, loss, accuracy
 
         count += args.batchsize
-        print('{} / {}'.format(count, len(dataset)), end='\r', file=sys.stderr)
-        sys.stderr.flush()
+        print('{} / {}'.format(count, len(dataset)), end='\r')
```
Comment (kmaehashi) [inline examples/modelzoo/evaluate_caffe_net.py]: `end='\r'` also needs to be removed.

Comment (niboshi) [reply examples/modelzoo/evaluate_caffe_net.py]: Ah, it seems we should replace thsese `print`s with `sys.stdout.write`.

**Comment by niboshi** [comment]
Fixed.

**Comment by kmaehashi** [comment]
LGTM!

**Pull request merged**